ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/CollectionSelection.tsx

```diff
@@ -73,6 +79,17 @@ function CollectionSelection({
         COLLECTION_PAGE_SIZE
     );
 
+    // If there is an existing collection selected, fetch the details for it. This allows
+    // us to display the collection name in the dropdown even if the collection is not
+    // in the current page of results.
+    const selectedCollectionFetch = useCallback(() => {
+        if (selectedScope?.id) {
+            return getCollection(selectedScope?.id).request;
+        }
+        return Promise.resolve(undefined);
+    }, [selectedScope?.id]);
```

Comment:
This is a good quick fix for the issue. The only thing is that it'll re-fetch using this API whenever the user selects a scope, even when the collection is already in memory (ie. the list of collections that were fetched). We could do an additional check for that, but in my opinion, it probably isn't a big deal if we're going to rethink this flow with the UX team.